src/LibraryManager.Vsix/Shared/Telemetry.cs

```diff
@@ -46,5 +50,135 @@ public static void TrackException(string name, Exception exception)
+                        }
+
+                        if (result.UpToDate)
+                        {
+                            uptodateProviderResults.Add(result);
+                        }
+
+                        if (result.Cancelled)
+                        {
+                            cancelledProviderResults.Add(result);
+                        }
+                    }
+                }
+
+                if (successfulProviderResults.Count > 0)
+                {
+                    telResult[$"LibrariesCount_{provider}_Success"] = successfulProviderResults.Count;
+                    telResult[$"LibrariesIDs_{provider}_Success"] = string.Join(":", GetHashedLibrariesNames(successfulProviderResults));
+                }
+
+                if (failedProviderResults.Count > 0)
+                {
+                    List<string> providerErrorCodes = GetErrorCodes(failedProviderResults);
+
+                    telResult[$"LibrariesCount_{provider}_Failure"] = failedProviderResults.Count;
+                    telResult[$"LibrariesIDs_{provider}_Failure"] = string.Join(":", GetHashedLibrariesNames(failedProviderResults));
+                    telResult.Add($"ErrorCodes_{provider}", string.Join(":", providerErrorCodes));
+                }
+
+                if (cancelledProviderResults.Count > 0)
+                {
+                    telResult[$"LibrariesCount_{provider}_Cancelled"] = cancelledProviderResults.Count;
+                    telResult[$"LibrariesIDs_{provider}_Cancelled"] = string.Join(":", GetHashedLibrariesNames(cancelledProviderResults));
+                }
+
+                if (uptodateProviderResults.Count > 0)
+                {
+                    telResult[$"LibrariesCount_{provider}_Uptodate"] = uptodateProviderResults.Count;
+                    telResult[$"LibrariesIDs_{provider}_Uptodate"] = string.Join(":", GetHashedLibrariesNames(uptodateProviderResults));
+                }
+            }
+
+            TrackUserTask($@"{operation}_Operation", TelemetryResult.None, telResult.Select(i => new KeyValuePair<string, object>(i.Key, i.Value)).ToArray());
+        }
+
+        internal static void LogErrors(string eventName, IEnumerable<ILibraryOperationResult> results)
+        {
+            List<string> errorCodes = GetErrorCodes(results.Where(r => r.Errors.Any()));
+            TrackUserTask(eventName, TelemetryResult.Failure, new KeyValuePair<string, object>("Errorcode", string.Join(":", errorCodes)));
+        }
+
+        private static List<string> GetErrorCodes(IEnumerable<ILibraryOperationResult> results)
+        {
+            List<string> errorCodes = new List<string>();
+
+            foreach (ILibraryOperationResult result in results)
+            {
+                errorCodes.AddRange(result.Errors.Select(e => e.Code));
+            }
+
+            return errorCodes;
+        }
+
+        private static List<string> GetHashedLibrariesNames(IEnumerable<ILibraryOperationResult> results)
```

Comment:
Please note use of own hashing method so we can concatenate the hashed values of individual libraries